dGame/dComponents/TriggerComponent.cpp

```diff
@@ -242,51 +249,53 @@ void TriggerComponent::HandlePushObject(Entity* targetEntity, std::vector<std::s
 
 void TriggerComponent::HandleRepelObject(Entity* targetEntity, std::string args){
 	auto* phantomPhysicsComponent = m_Parent->GetComponent<PhantomPhysicsComponent>();
-	if (!phantomPhysicsComponent) return;
+	if (!phantomPhysicsComponent) {
+		Game::logger->Log("TriggerComponent::HandleRepelObject", "Phantom Physics component not found!");
+		return;
+	}
 	float forceMultiplier;
 	GeneralUtils::TryParse<float>(args, forceMultiplier);
 	phantomPhysicsComponent->SetPhysicsEffectActive(true);
-	phantomPhysicsComponent->SetEffectType(1); // repel/repulse
+	phantomPhysicsComponent->SetEffectType(ePhysicsEffectType::REPULSE);
 	phantomPhysicsComponent->SetDirectionalMultiplier(forceMultiplier);
 
 	auto triggerPos = m_Parent->GetPosition();
 	auto targetPos = targetEntity->GetPosition();
 
 	// normalize the vectors to get the direction
 	auto delta = targetPos - triggerPos;
-	auto abs = sqrtf(delta.x * delta.x + delta.y * delta.y + delta.z * delta.z);
-	NiPoint3 direction = NiPoint3::ZERO;
-	direction.x = delta.x / abs;
-	direction.y = delta.y / abs;
-	direction.z = delta.z / abs;
-
+	auto abs = delta.Length();
+	NiPoint3 direction = delta / abs;
 	phantomPhysicsComponent->SetDirection(direction);
 
 	EntityManager::Instance()->SerializeEntity(m_Parent);
 }
 
 void TriggerComponent::HandlePlayCinematic(Entity* targetEntity, std::vector<std::string> argArray) {
-	// [cinematic name],(lead-in in seconds),(“wait” to wait at end),(“unlock” to NOT lock the player controls),(“leavelocked” to leave player locked after cinematic finishes),(“hideplayer” to make player invisible during cinematic)
 	float leadIn = -1.0;
-	if (argArray.size() >= 2) GeneralUtils::TryParse<float>(argArray.at(1), leadIn);
-	int wait = 0;
-	if (argArray.size() >= 3 && argArray.at(2) == "wait") wait = 1;
-	bool unlock;
-	if (argArray.size() >= 4 && argArray.at(3) == "unlock") unlock = false;
+	auto wait = eEndBehavior::RETURN;
+	bool unlock = true;
 	bool leaveLocked = false;
-	if (argArray.size() >= 5 && argArray.at(4) == "leavelocked") leaveLocked = true;
 	bool hidePlayer = false;
-	if (argArray.size() >= 6 && argArray.at(5) == "hideplayer") hidePlayer = true;
-	GameMessages::SendPlayCinematic(targetEntity->GetObjectID(), GeneralUtils::UTF8ToUTF16(argArray.at(0)), targetEntity->GetSystemAddress(), true, true, false, false, 0, hidePlayer, leadIn, leaveLocked, unlock);
+
+	if (argArray.size() >= 2) GeneralUtils::TryParse<float>(argArray.at(1), leadIn);
+	else if (argArray.size() >= 3 && argArray.at(2) == "wait") wait = eEndBehavior::WAIT;
+	else if (argArray.size() >= 4 && argArray.at(3) == "unlock") unlock = false;
+	else if (argArray.size() >= 5 && argArray.at(4) == "leavelocked") leaveLocked = true;
+	else if (argArray.size() >= 6 && argArray.at(5) == "hideplayer") hidePlayer = true;
```

Comment:
was poor phrasing on my end on what change I was thinking about here - 
I was more thinking of something akin to what we see in the client when reading packets
```
if () {
    if () {
        if () {
        }
    }
}
```
Sorry about that!  Right now this ignores all elements past the first one

Reply:
oh yeah, that's my bad too, this code just doesn't work at all